sklearn/random_projection.py

```diff
@@ -217,7 +219,8 @@ def _sparse_random_matrix(n_components, n_features, density='auto',
     n_features : int,
         Dimensionality of the original source space.
 
-    density : float in range ]0, 1] or 'auto', optional (default='auto')
+    density : float or 'auto', default='auto'
+        float in range (0,1]
         Ratio of non-zero component in the random projection matrix.
```

Comment:
```suggestion
        Ratio of non-zero component in the random projection matrix in the
        range `(0, 1]`
```

Reply:
Would this be better?
`eps : float, default=0.1
        Parameter to control the quality of the embedding according to
        the Johnson-Lindenstrauss lemma when n_components is set to
        'auto'. Needs to be strictly positive.
`